packages/node-app/src/routes/giveaway.route.ts

```diff
@@ -26,11 +26,10 @@ const upload = multer({ storage });
 export const router = express.Router();
 
 router.get('/', listGiveaways);
+router.post('/', verifyAdmin, upload.single('image'), createGiveaway);
 router.get('/:id', getGiveaway);
+router.put('/:id', verifyAdmin, upload.single('image'), updateGiveaway);
 router.put('/:id/participants', addParticipant);
-router.get('/:id/participants', getParticipants);
+router.get('/:id/participants', verifyAdmin, getParticipants);
```

Comment:
Currently we need this route to be open to the users as well, it is used on the frontend to several tasks such as:
- get user participation state
- check if should be notified about winning
- get the names of the winners
- calculate the winning chance

These could now be optimized since we have access to the user that made the request, we should create tasks for those.

Reply:
Thanks for catching that 🙏 Created a ticket to fix that in the future: https://runtime-revolution.atlassian.net/browse/WEB3-65